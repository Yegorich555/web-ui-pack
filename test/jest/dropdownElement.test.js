import { WUPDropdownElement } from "web-ui-pack";
// import * as h from "../testHelper";

/** @type WUPDropdownElement */
let el;

beforeEach(() => {
  !WUPDropdownElement && console.error("missed");
});

afterEach(() => {
  document.body.innerHTML = "";
  jest.restoreAllMocks();
});

describe("dropdownElement", () => {
  test("basic usage", async () => {
    // todo implement
  });
});
