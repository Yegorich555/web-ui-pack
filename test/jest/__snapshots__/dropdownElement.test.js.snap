// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dropdownElement common tests render + styles 1`] = `<wup-dropdown />`;

exports[`dropdownElement common tests render + styles 2`] = `
<style>
  .wup-hidden, [wup-hidden] {
position: absolute;
height:1px; width:1px;
top:0;left:0;
overflow:hidden;
clip:rect(1px,1px,1px,1px);
min-width:initial;
padding:0;}
  :root {
          --base-bg: #fff;
          --base-text: #232323;
          --base-focus: #00778d;
          --base-btn-bg: #009fbc;
          --base-btn-text: #fff;
          --base-btn-focus: #005766;
          --base-btn2-bg: var(--base-btn-text);
          --base-btn2-text: var(--base-btn-bg);
          --base-btn3-bg: var(--base-bg);
          --base-btn3-text: inherit;
          --base-sep: #e4e4e4;
          --border-radius: 6px;
          --anim-time: 200ms;
          --anim: var(--anim-time) cubic-bezier(0, 0, 0.2, 1) 0ms;
        }
  
      WUP-DROPDOWN {
        contain: style;
        display: inline-block;
      }
WUP-DROPDOWN button {
  box-shadow: none;
  border: 1px solid var(--base-btn-bg);
  border-radius: var(--border-radius);
  box-sizing: border-box;
  padding: 0.5em;
  margin: 1em 0;
  min-width: 10em;
  cursor: pointer;
  font: inherit;
  font-weight: bold;
  background: var(--base-btn-bg);
  color: var(--base-btn-text);
  outline: none;
}
WUP-DROPDOWN button:focus {
  border-color: var(--base-btn-focus);
}
@media (hover: hover) and (pointer: fine) {
  WUP-DROPDOWN button:hover {
    box-shadow: inset 0 0 0 99999px rgba(0,0,0,0.2);
  }
}
WUP-DROPDOWN button[disabled] {
  opacity: 0.3;
  cursor: not-allowed;
  -webkit-user-select: none;
  user-select: none;
}
WUP-DROPDOWN button[aria-busy] {
  cursor: wait;
}
      WUP-DROPDOWN button {
        min-width: initial;
        margin: 0;
        padding: 0.7em;
      }
WUP-DROPDOWN [menu] {
  padding: 0;
}
WUP-DROPDOWN [menu] ul {
  margin: 0;
  padding: 0;
  list-style-type: none;
  cursor: pointer;
  overflow: auto;
  max-height: 300px;
}

WUP-DROPDOWN [menu] ul::-webkit-scrollbar {
  width: 10px; height: 10px;
  cursor: pointer;
}
WUP-DROPDOWN [menu] ul::-webkit-scrollbar-corner {
  background: none;
  cursor: pointer;
}
WUP-DROPDOWN [menu] ul::-webkit-scrollbar-thumb {
  border: 3px solid rgba(0,0,0,0);
  background-clip: padding-box;
  background-color: var(--scroll, rgba(0,0,0,0.2));
  border-radius: 999px;
  cursor: pointer;
}
WUP-DROPDOWN [menu] ul::-webkit-scrollbar-track-piece:vertical:start,
WUP-DROPDOWN [menu] ul::-webkit-scrollbar-track-piece:vertical:end,
WUP-DROPDOWN [menu] ul::-webkit-scrollbar-track-piece:horizontal:start,
WUP-DROPDOWN [menu] ul::-webkit-scrollbar-track-piece:horizontal:end {
  margin: 0;
  cursor: pointer;
}
@media (hover) {
  WUP-DROPDOWN [menu] ul::-webkit-scrollbar-thumb:hover {
    background-color: var(--scroll-hover, rgba(0,0,0,0.5));
    cursor: pointer;
  }
}
WUP-DROPDOWN [menu] li {
  padding: 1em;
  border-radius: var(--border-radius);
}
@media (hover: hover) and (pointer: fine) {
  WUP-DROPDOWN [menu] li:hover {
    box-shadow: inset 0 0 3px 0 var(--base-focus);
  }
}
WUP-DROPDOWN [menu] li[aria-selected="true"] {
  color: var(--ctrl-selected);
  display: flex;
}
WUP-DROPDOWN [menu] li[aria-selected="true"]:after {
  content: "";
  --ctrl-icon-img: var(--wup-icon-check);
  
display: inline-block;
width: var(--ctrl-icon-size);
min-height: var(--ctrl-icon-size);
box-sizing: content-box;
margin: 0;
padding: 5px;
flex: 0 0 auto;
align-self: stretch;
border: none;
box-shadow: none;
background: var(--ctrl-icon);
-webkit-mask-size: var(--ctrl-icon-size);
mask-size: var(--ctrl-icon-size);
-webkit-mask-repeat: no-repeat;
mask-repeat: no-repeat;
-webkit-mask-position: center;
mask-position: center;
-webkit-mask-image: var(--ctrl-icon-img);
mask-image: var(--ctrl-icon-img);
  background: var(--ctrl-selected);
  margin-left: auto;
  padding: 0;
}
WUP-DROPDOWN [menu] li[focused] {
  box-shadow: inset 0 0 4px 0 var(--base-focus);
}
</style>
`;
